google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/GetProductBiddingCategoryConstant.java

```diff
@@ -0,0 +1,193 @@
+import com.google.ads.googleads.v1.services.SearchGoogleAdsRequest;
+import com.google.api.client.util.Preconditions;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+/** This example fetches the set of valid ProductBiddingCategories. */
+public class GetProductBiddingCategoryConstant {
+
+  private static final int PAGE_SIZE = 1_000;
+
+  private static class GetProductBiddingCategoryConstantParams extends CodeSampleParams {
+
+    @Parameter(names = ArgumentNames.CUSTOMER_ID, required = true)
+    Long customerId;
+  }
+
+  public static void main(String args[]) {
+    GetProductBiddingCategoryConstantParams params = new GetProductBiddingCategoryConstantParams();
+    if (!params.parseArguments(args)) {
+      // Either pass the required parameters for this example on the command line, or insert them
+      // into the code here. See the parameter class definition above for descriptions.
+      params.customerId = Long.parseLong("ENTER_CUSTOMER_ID_HERE");
+    }
+
+    GoogleAdsClient googleAdsClient;
+    try {
+      googleAdsClient = GoogleAdsClient.newBuilder().fromPropertiesFile().build();
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new GetProductBiddingCategoryConstant().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   */
+  private static void runExample(GoogleAdsClient googleAdsClient, long customerId) {
+    // Creates the query.
+    String query =
+        "SELECT "
+            + "product_bidding_category_constant.localized_name, "
+            + "product_bidding_category_constant.product_bidding_category_constant_parent "
+            + "FROM "
+            + "product_bidding_category_constant "
+            + "WHERE "
+            + "product_bidding_category_constant.country_code IN ('US')";
```

Comment:
Explain where these constants come from, e.g. reference docs.

Reply:
I don't think this resource exists in GAAPI (it is in AdWords). I have filed a bug internally for this and can update once this list is released.

Reply:
Is it not this: https://developers.google.com/google-ads/api/docs/fields/language_constant?

Reply:
I don't know if that is the exact same thing because the language_code field on ProductBiddingCategoryConstant needs to be a string, but the structure of it is unspecified. For example, [this resource](https://developers.google.com/adwords/api/docs/appendix/codes-formats#country-codes) exists for AdWords but not yet for GAAPI.